src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BulldozerProducer bulldozerProducer;
+    private final ExcavatorProducer excavatorProducer;
+    private final TruckProducer truckProducer;
+
+    public MachineServiceImpl() {
+        bulldozerProducer = new BulldozerProducer();
+        excavatorProducer = new ExcavatorProducer();
+        truckProducer = new TruckProducer();
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                return new ArrayList<>(bulldozerProducer.get());
+            case "Excavator":
+                return new ArrayList<>(excavatorProducer.get());
+            case "Truck":
+                return new ArrayList<>(truckProducer.get());
```

Comment:
I'd replace it with the strategy `Map<Class<? extends Machine>, MachineProducer<? extends Machine>>` instead of switch based on class name string